### PR TITLE
add: Docs on running the Soccer Sim automatically

### PR DESCRIPTION
**Pull request by mrshu**
* Add docs on how to run the soccer sim automatically with a brief
  interlude into how the referee works.

* Normalize the environment variables Soccer Sim responds to.

* Document all the environment variables that are supported.

Signed-off-by: mr.Shu <mr@shu.io>

### CODE REVIEW

File: controllers/rcj_soccer_referee_supervisor/rcj_soccer_referee_supervisor.py [hunk 1]
```diff
@@ -38,24 +38,32 @@ def output_path(
...
-TEAM_YELLOW_ID = os.environ.get("TEAM_YELLOW_ID", "The Yellows")
-TEAM_YELLOW_INITIAL_SCORE = int(os.environ.get("TEAM_Y_INITIAL_SCORE", "0") or "0")
-TEAM_BLUE = os.environ.get("TEAM_BLUE_NAME", "The Blues")
-TEAM_BLUE_ID = os.environ.get("TEAM_BLUE_ID", "The Blues")
-TEAM_BLUE_INITIAL_SCORE = int(os.environ.get("TEAM_B_INITIAL_SCORE", "0") or "0")
-MATCH_ID = os.environ.get("MATCH_ID", 1)
-HALF_ID = os.environ.get("HALF_ID", 1)
-REC_FORMATS = [f for f in os.environ.get("REC_FORMATS", "").split(",") if f]
-MATCH_TIME = int(os.environ.get("MATCH_TIME", DEFAULT_MATCH_TIME))
+TEAM_YELLOW = os.environ.get("RCJ__SIM_TEAM_YELLOW_NAME", "The Yellows")
```
Comment (Adman) [inline controllers/rcj_soccer_referee_supervisor/rcj_soccer_referee_supervisor.py:46]:
```suggestion
TEAM_YELLOW = os.environ.get("RCJ_SIM_TEAM_YELLOW_NAME", "The Yellows")
```

File: docs/docs/how_to_run_sim.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+having the code for two teams and the RoboCupJunior Soccer Sim code to getting
+an output in form of either an MPEG-4 video or a HTML site.
+
+**Input**
+- Source code for the yellow team
+- Source code for the blue team
+- RoboCupJunior Soccer Sim
+
+**Output**
+- MPEG-4 video or HTML site
```
Comment (Adman) [inline docs/docs/how_to_run_sim.md:17]:
```suggestion
- MPEG-4 video and/or HTML site
- JSON file containing all the important events happened during the game
```

File: docs/docs/how_to_run_sim.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+## Preliminaries
+
+This guide makes a couple assumptions:
```
Comment (Adman) [inline docs/docs/how_to_run_sim.md:21]: Not sure if "of" is correct to be added.
```suggestion
This guide makes a couple of assumptions:
```

File: docs/docs/how_to_run_sim.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+1. You use a UNIX-like environment (i.e. something like Linux or macOS)
```
Comment (Adman) [inline docs/docs/how_to_run_sim.md:22]:
```suggestion
1. You use an UNIX-like environment (i.e. something like Linux or macOS)
```

File: docs/docs/how_to_run_sim.md (hunk 3 above)
Comment (mrshu) [reply docs/docs/how_to_run_sim.md:22]: Makes sense -- thanks @Adman !

File: docs/docs/how_to_run_sim.md (hunk 4 above)
Comment (mrshu) [reply docs/docs/how_to_run_sim.md:23]: @Adman I guess I don't care about that that much :slightly_smiling_face:

**Comment by mrshu** [comment]
Thanks @Adman -- would appreciate any feedback as to whether this makes sense to you.

**Comment by Adman** [comment]
@mrshu yes, makes sense. Maybe we could mention how to play two halftimes.

**Review by Adman — APPROVED** [review]

**Comment by felipenmartins** [comment]
I think this is very good! I did not try to replicate it, though.

**Comment by mrshu** [comment]
Thanks @Adman, I will try to add a section on Dockerization in a further pass as well

**Pull request merged**